libreyolo/label/server.py

```
@@ -0,0 +1,587 @@
+    bound eagerly (an in-use port raises ``OSError`` to retry). Returns
+    ``(httpd, url, session)`` (``session`` is ``None`` in home mode).
+    """
+    session = DatasetSession(data) if data else None
+    state = _LabelState(session, device=device, assist=assist)
+    state.host = host
+    state.port = port
+    if session is not None:
+        state.register_current(data)
+    handler = type("BoundLabelHandler", (_Handler,), {"state": state})
+    httpd = ThreadingHTTPServer((host, port), handler)
+    url = "http://%s:%d" % (host, port)
```

Ruff (0.15.15)

[warning] Use format specifiers instead of percent format

(UP031)